site-src/geps/gep-1282.md

```diff
@@ -98,12 +98,302 @@ Of course, from the section above, you can see that there are many other feature
+capabilities may be added as needed in subsequent versions of Gateway API.  BackendCapabilities also contains a
+reference to a backend service name and type, and a BackendCapabilitiesStatus.
+
+In https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1beta1/shared_types.go add a new type:
+```go
+// BackendCapabilities references a Kubernetes Service and defines any backend 
+// capabilities that Service requires.  
+//
+// +optional
+type BackendCapabilities struct {
+    // BackendName is the name of the backend service to which these backend 
+    // capabilities apply.
+    // +required
+    BackendName string `json:”backendName”`
+
+    // BackendType is the type of backend to which these backend 
+    // capabilities apply.
+    //
+    // +required
+    BackendType BackendTargetRef `json:”backendType”`
+
+	// Spec specifies the capabilities supported by this
+    // Service wrapper, if any.  The capabilities are predefined, and have two
+    // characteristics in common: they are tightly bound to the backend, 
+    // rather than being important at the route level, and the service owner
+    // (application developer in the Gateway API personas) should control 
+    // these settings, not the owner of the Gateway implementation (the 
+    // Cluster Operator in the Gateway API personas).
+    // 
+    // Support: Extended
+	//
+    // +required
+	Spec BackendCapabilitiesSpec `json:”spec”`
+
+	// Status represents the current state of the backend and its capabilities.
+    // +optional
+	Status *BackendCapabilitiesStatus `json:”status,omitempty”`
+}
+
+// BackendTargetReference identifies an API object to apply BackendCapabilities 
+// to. 
+// 
+type BackendTargetReference struct {
+	// Group is the group of the target resource.
+	Group Group `json:"group"`
+
+	// Kind is kind of the target resource.
+    // Valid values include:
+    // * "Service" (the default if no Kind is specified)
+    // * "ServiceImport"
+    //
+    // Invalid values include:
+    //
+    // * "invalid/kind" - "/" is an invalid character
+    // +required
+	Kind string `json:"kind"`
+
+	// Version is the version of the target resource.
+	Version ObjectName `json:"version"`
+
+	// Name is the name of the target resource.
+	Name ObjectName `json:"name"`
+}
+
+// BackendCapabilitiesSpec defines a group of specific properties held by its
+// affiliated BackendCapabilitiesService, and specified by the Service owner 
+// (typically the application developer).  Examples of properties that 
+// belong in the BackendCapabilitiesSpec include: TLS re-encryption, session
+// state, timeouts, retries, Websockets, HTTP/2, load balancer algorithm, 
+// healthcheck info, external traffic policy, connection limits, error page,
+// protocol switch, alerts, and others. 
+//
+// Support: Extended
+//
+type BackendCapabilitiesSpec struct {
+
+	// +optional
+	Reencrypt *TLSReencryptType `json:”reencrypt,omitempty”`
```

Comment:
I see this was also discussed in https://github.com/kubernetes-sigs/gateway-api/discussions/1285#discussioncomment-3261405. To me `reencrypt` is quite misleading since, as said, it is just about TLS for backend/upstream hop. Personally, I would rather vote something neutral like `tls`, with a description that it can be used also for re-encrypt use case?

Reply:
in https://gateway-api.sigs.k8s.io/guides/tls/?h=tls#clientserver-and-tls, since we call the first hop `Terminate`, how about calling the second hop `Extend` or `Forward`?  

I hesitate to call it `tls` because we already use that in `Gateway.spec.Listeners` https://github.com/kubernetes-sigs/gateway-api/blob/96414f1ce72b2277cde5cb8e8abcd97ea039fe4a/apis/v1beta1/gateway_types.go#L207 and documented in https://gateway-api.sigs.k8s.io/guides/tls/?h=tls#clientserver-and-tls

Reply:
I used TLS, and updated some more of the godoc.  Please take a look.